src/plans/conjugate_gradient_plan.jl

```diff
@@ -1,3 +1,21 @@
+
+struct DirectionUpdateRuleStorage{TC<:DirectionUpdateRule,TStorage<:StoreStateAction}
+    coefficient::TC
+    storage::TStorage
+end
+
+function DirectionUpdateRuleStorage(
+    M::AbstractManifold, dur::DirectionUpdateRule, p=rand(M), X=zero_vector(M, p)
+)
+    ursp = update_rule_storage_points(dur)
+    ursv = update_rule_storage_vectors(dur)
+    # StoreStateAction makes a copy
+    sa = StoreStateAction(
+        Symbol[], NamedTuple{ursp}(map(x -> p, ursp)), NamedTuple{ursv}(map(x -> X, ursv))
```

Comment:
already more like a detail, should it maybe be `x->copy(M, p)` since we want different memory for every point we store?

Reply:
ah StoreAction makes a copy, sorry, missed that – one question is of course where the copying fits better here or in the constructor.

Reply:
Yes, I've even made a comment about that :slightly_smiling_face: . I'm not sure where the copy would be more fitting but making it in `StoreStateAction` seems safer so that's why I've chosen that.

Reply:
Ok, sure, hiding it from the user is maybe safer than requiring the user to do copies, when we anyways need them.